### ISSUE TITLE
The code generator should allow swapping of cmp operands

### ISSUE THREAD

**Issue by CarolEidt** [dotnet/coreclr#7270]
See the code in `Lowering::TreeNodeInfoInitCmp()` and `CodeGen::genCompareInt()`. If op1 is a constant, it should be able to swap them to generate `cmp R/M, imm`.

**Comment by cmckinsey** [comment]
Perhaps this would be a good up for grabs? @CarolEidt

**Comment by mikedn** [comment]
Maybe this should wait for #7255?

**Comment by CarolEidt** [comment]
I don't know for certain (which is why I deferred doing this instead of incorporating it into my PR), but it's my guess that this is a vestige of non-LIR constraints. I don't think it is dependent on #7255, but I could be mistaken.

**Comment by mikedn** [comment]
>  I don't think it is dependent on #7255, but I could be mistaken.

It's not dependent but 7255 looks like a significant change in the same area. Anyway, I'll take a quick look to see what needs to be done to get this to work.

**Comment by mikedn** [comment]
Do you have any test case for this issue? I tried a few things but it seems that `gtSetEvalOrder` swaps the compare operands so that the constant is in the second position. I suppose we may have a problem if a constant is produced by optimizations and then the order isn't changed appropriately but the various things I've tried either generate `cmp R/M, imm` or don't but for a different reason.

For example, `return GetStaticField() < GetConstant();` generates:
```
       A13046A702   mov      eax, dword ptr [classVar[0x2a74e60]]
       83F82A       cmp      eax, 42
```
but that's because inlining introduces a lclvar to hold the value of the static field.

**Comment by mikedn** [comment]
This issue and the associated code comments do not appear to be useful. There are only 3 instances of `relop icon, x` in the entire framework and they're all of the form `eq/ne 0, 0`. I'm not sure why they aren't folded (they're produced by `default(T) != null` when `T` is a reference type) but swapping doesn't help anyway.

I'd say that we should simply remove the comments from the code (and try to fix `default(T) != null` separately).
